creature_factory.cpp

```diff
@@ -1064,6 +1064,30 @@ CreatureAttributes CreatureFactory::getAttributesFromId(CreatureId id) {
           c.maxLevelIncrease[ExperienceType::SPELL] = 8;
           c.maxLevelIncrease[ExperienceType::ARCHERY] = 8;
       );
+    case CreatureId::UNICORN:
+      return CATTR(
+        c.viewId = ViewId::UNICORN;
+        c.attr = LIST(20_dam, 20_def, 20_spell_dam, 200_spd );
+        c.body = Body::nonHumanoid(Body::Size::LARGE).setWeight(500).setHorseBodyParts();
+        //Ideally, you should club them to death or chop them up with a sword.
+        c.permanentEffects[LastingEffect::RANGED_RESISTANCE] = 1;
+        c.permanentEffects[LastingEffect::MAGIC_RESISTANCE] = 1;
+        c.barehandedAttack = AttackType::HIT;
+        c.courage = 100;
+        //They heal up and summon friends.
+        c.spells->add(SpellId::HEAL_SELF);
+        c.spells->add(SpellId::HEAL_OTHER);
+        c.spells->add(SpellId::SUMMON_SPIRIT);
+        c.chatReactionFriendly = "\"mhhhhhrrrr!\""_s;
+        c.chatReactionHostile = "\"mhhhhhrrrr!\""_s;
+        c.name = "unicorn";
+        //Pet names like dogs would have.
+        c.name->setFirst(NameGenerator::get(NameGeneratorId::DOG)->getNext());
+        c.name->setGroup("herd");
+        c.innocent = true;
```

Comment:
You don't want to set them as innocent, this is something like civilians, you don't need to kill them to conquer the village. (I'm not sure what the game will do if everyone in the village is innocent).